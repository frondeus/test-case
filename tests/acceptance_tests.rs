#![cfg(test)]

mod acceptance {
    use itertools::Itertools;
    use std::env;
    use std::path::PathBuf;
    use std::process::Command;

    #[test]
    fn basic() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("basic"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn hamcrest_assertions() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("hamcrest_assertions"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn r#async() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("async"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }
}
