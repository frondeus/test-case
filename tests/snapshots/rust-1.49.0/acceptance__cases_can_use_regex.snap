---
source: tests/acceptance_tests.rs
assertion_line: 136
expression: output
---
    /
    ^
    let re = ::test_case::Regex::new(r#"/d{4}"#).expect("Regex::new");
    let re = ::test_case::Regex::new(r#"abc"#).expect("Regex::new");
    re.is_match(_result)
    re.is_match(_result)
    regex_test::_abcabc201_expects_complex_regex_r_d_4_
    regex_test::_kumkwat_expects_complex_regex_r_
    regex_test::_kumkwat_expects_complex_regex_r_abc_
)', src/lib.rs:5:1
---- regex_test::_abcabc201_expects_complex_regex_r_d_4_ stdout ----
---- regex_test::_kumkwat_expects_complex_regex_r_ stdout ----
---- regex_test::_kumkwat_expects_complex_regex_r_abc_ stdout ----
error: incomplete escape sequence, reached end of pattern prematurely
error: test failed, to rerun pass '--lib'
failures:
failures:
regex parse error:
running 6 tests
test regex_test::_abcabc201_expects_complex_regex_r_d_ ... ok
test regex_test::_abcabc201_expects_complex_regex_r_d_4_ ... FAILED
test regex_test::_abcabc_expects_complex_regex_r_abc_ ... ok
test regex_test::_kumkwat_expects_complex_regex_r_ ... FAILED
test regex_test::_kumkwat_expects_complex_regex_r_abc_ ... FAILED
test regex_test::_kumkwat_expects_complex_regex_r_kumkwat_ ... ok
test result: FAILED. 3 passed; 3 failed; 0 ignored; 0 measured; 0 filtered out
thread 'regex_test::_abcabc201_expects_complex_regex_r_d_4_' panicked at 'assertion failed: {
thread 'regex_test::_kumkwat_expects_complex_regex_r_' panicked at 'Regex::new: Syntax(
thread 'regex_test::_kumkwat_expects_complex_regex_r_abc_' panicked at 'assertion failed: {
}', src/lib.rs:5:1
}', src/lib.rs:5:1
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
